### ISSUE TITLE
B5 Compilation Error: ‘g_egl_dynProcs’ undeclared

### ISSUE THREAD

**Issue by TheAifam5** [gnif/LookingGlass#962]
Hey,

I am using Gentoo and maintain my Gentoo repository, while upgrading to the B5 with following CMake configuration, the compilation fails.

## CMake Output:
```
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/x86_64-pc-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/x86_64-pc-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_SUPPORTS_MARCH_NATIVE
-- Performing Test COMPILER_SUPPORTS_MARCH_NATIVE - Success
-- Found PkgConfig: /usr/bin/x86_64-pc-linux-gnu-pkg-config (found version "1.8.0") 
-- Checking for module 'fontconfig'
--   Found fontconfig, version 2.13.1
-- BFD libs: /usr/lib64/libbfd.so
-- Found BFD: /usr/lib64/libbfd.so  
-- BFD libs: /usr/lib64/libbfd.so
-- Checking for modules 'spice-protocol;nettle;hogweed'
--   Found spice-protocol, version 0.14.3
--   Found nettle, version 3.7.3
--   Found hogweed, version 3.7.3
-- Checking for modules 'x11;xi;xfixes;xscrnsaver;xinerama;xcursor;xpresent'
--   Found x11, version 1.7.3
--   Found xi, version 1.8
--   Found xfixes, version 6.0.0
--   Found xscrnsaver, version 1.2.3
--   Found xinerama, version 1.1.4
--   Found xcursor, version 1.2.0
--   Found xpresent, version 1.0.0
-- Checking for module 'gl'
--   Found gl, version 1.2
-- The following features have been enabled:

 * OPTIMIZE_FOR_NATIVE, Optimized for native
 * ENABLE_OPENGL, Legacy OpenGL renderer.
 * ENABLE_BACKTRACE, Backtrace support.
 * ENABLE_X11, X11 support.

-- The following features have been disabled:

 * ENABLE_EGL, EGL renderer.
 * ENABLE_ASAN, AddressSanitizer support.
 * ENABLE_UBSAN, UndefinedBehaviorSanitizer support.
 * ENABLE_WAYLAND, Wayland support.
 * ENABLE_LIBDECOR, libdecor support.
```

## Compilation error:
```
FAILED: CMakeFiles/looking-glass-client.dir/src/eglutil.c.o 
/usr/bin/x86_64-pc-linux-gnu-gcc -DCIMGUI_DEFINE_ENUMS_AND_STRUCTS=1 -DIMGUI_DISABLE_OBSOLETE_FUNCTIONS=1 -DIMGUI_IMPL_API="extern	\"C\"	" -I/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/client/include -I/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/client_build/include -I/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/common/include -I/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/repos/LGMP/lgmp/include -I/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/repos/PureSpice/include -I/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/repos/cimgui -I/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/repos/cimgui/imgui -isystem /usr/include/freetype2 -isystem /usr/include/harfbuzz -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/uuid  -march=native -ggdb -O2 -pipe   -D ENABLE_OPENGL -D ATOMIC_LOCKING -D GL_GLEXT_PROTOTYPES -march=native -Wall -Wextra -Wno-sign-compare -Wno-unused-parameter -Wimplicit-fallthrough=2 -Werror -Wfatal-errors -ffast-math -fdata-sections -ffunction-sections -std=gnu11 -MD -MT CMakeFiles/looking-glass-client.dir/src/eglutil.c.o -MF CMakeFiles/looking-glass-client.dir/src/eglutil.c.o.d -o CMakeFiles/looking-glass-client.dir/src/eglutil.c.o -c /var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/client/src/eglutil.c
/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/client/src/eglutil.c: In function ‘swapWithDamageInit’:
/var/tmp/portage/app-emulation/looking-glass-0_beta5/work/looking-glass-0_beta5/client/src/eglutil.c:32:66: error: ‘g_egl_dynProcs’ undeclared (first use in this function); did you mean ‘egl_dynProcsInit’?
   32 |   if (util_hasGLExt(exts, "EGL_KHR_swap_buffers_with_damage") && g_egl_dynProcs.eglSwapBuffersWithDamageKHR)
      |                                                                  ^~~~~~~~~~~~~~
      |                                                                  egl_dynProcsInit
compilation terminated due to -Wfatal-errors.
```

Best regards,
TheAifam5

**Comment by gnif** [comment]
Not a bug, you're building without EGL support. Please follow the instructions and join one of the communities to discuss your problem.

**Comment by TheAifam5** [comment]
I do build without EGL support because on my machine the EGL is much worse than OGL, since I do not need EGL as a backend I disabled it. Since B5 there is now a hard dependency on EGL when only OGL is enabled, on B4 that was not a case - that's the reason I made this bug report because of the behavior change.

If OGL now requires EGL (that's why compilation fails, because ENABLE_EGL is not defined for OGL-only comilation, see here: [client/include/egl_dynprocs.h](https://github.com/gnif/LookingGlass/blob/952ebea2c5c349a45bb6b5c4c6b44cd68dbf67ca/client/include/egl_dynprocs.h#L23)) a additional check should be performed at CMakeFile level to ensure that OGL support can be only enabled when EGL has been enabled.
